File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.cpp
class Solution {
public:
    int maxAdjacentDistance(vector<int>& nums) {
        int maxDiff = abs(nums[0]-nums[nums.size()-1]);
        for(int i =0; i <nums.size()-1;++i){
           maxDiff=max(maxDiff, abs(nums[i+1]-nums[i]));
        }
        return maxDiff;
    }
};

File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.cs
public class Solution {
    public int MaxAdjacentDistance(int[] nums) {
        var maxDiff = 0;

        for (int i = 1; i < nums.Length; i++)
            maxDiff = Math.Max(Math.Abs(nums[i] - nums[i - 1]), maxDiff);
        
        return Math.Max(maxDiff, Math.Abs(nums[^1] - nums[0]));   
    }
}